File: tests/MiniZinc.SolveTests/Logging.fs
namespace MiniZinc.Tests

open Microsoft.Extensions.Logging
open Serilog

[<AutoOpen>]
module Logging =
    
    let logger =
            
        let serilogLogger =
            LoggerConfiguration()
                .WriteTo.Console()
                .CreateLogger()
                
        let factory =
            new LoggerFactory()
        
        factory
            .AddSerilog(serilogLogger)
            .CreateLogger("MiniZinc")

File: tests/MiniZinc.Net.Tests/Logging.cs
using Microsoft.Extensions.Logging;

namespace MiniZinc.Tests;

public static class Logging
{
    private static readonly ILoggerFactory _factory = LoggerFactory.Create(b => b.AddSimpleConsole(x => x.SingleLine = true));

    public static ILogger Create<T>() => _factory.CreateLogger<T>();
}
